stratum/hal/lib/barefoot/bf_sde_interface.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020-present Open Networking Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef STRATUM_HAL_LIB_BAREFOOT_BF_SDE_INTERFACE_H_
+#define STRATUM_HAL_LIB_BAREFOOT_BF_SDE_INTERFACE_H_
+
+#include <memory>
+
+#include "stratum/glue/integral_types.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/glue/status/statusor.h"
+#include "stratum/hal/lib/common/common.pb.h"
+#include "stratum/hal/lib/common/utils.h"
+#include "stratum/lib/channel/channel.h"
+
+namespace stratum {
+namespace hal {
+namespace barefoot {
+
+// TODO(max): docs
+class BfSdeInterface {
+ public:
+  // PortStatusEvent encapsulates the information received on a port status
+  // event. Port refers to the SDE internal device port ID.
+  struct PortStatusEvent {
+    int device;
```

Comment:
device refers to the sde device id. Should we consider translating these to be Stratum IDs?

Reply:
We kinda do already.

Check the calls to `SendPortOperStateGnmiEvent`, it uses the node ID: https://github.com/stratum/stratum/blob/12a01d535e8a0ace7e35e5ad215c9c172f8ab22d/stratum/hal/lib/barefoot/bf_chassis_manager.cc#L885

Unless you meant something else.

Reply:
I guess the problem is that the wrapper doesn't really know about Stratum IDs yet?

Reply:
`event.device` is used in one place only: `ReadPortStatusEvents` of the ChassisManger.

Reply:
What I was referring to is that we could maintain SDK port numbers entirely in the wrapper if we passed them through. This would keep these details hidden from callers.

Reply:
Not sure if the wrapper needs to know about P4rt node ids?

Reply:
We can do this later if we want/need to.